pyspecProcScripts/phasing.py

```diff
@@ -186,12 +186,54 @@ def costfun(ph1):
     # }}}
 
 
-def fid_from_echo(d, signal_pathway, fl=None, add_rising=False, fraction_nonnoise=0.1, direct="t2",
+def fid_from_echo(d, signal_pathway, fl=None, add_rising=False, direct="t2",
         exclude_rising=3,
         slice_multiplier=20,
         peak_lower_thresh=0.1,
         show_hermitian_sign_flipped=False,
         show_shifted_residuals=False):
+    """
+    Parameters
+    ==========
+    signal_pathway: dict
+                    coherence transfer pathways that correspond to the signal
+    fl:             figlist or None (default)
+                    If you want the diagnostic plots (showing the
+                    distribution of the data in the complex plane),
+                    set this to your figlist object.
+    add_rising:     boolean
+                    Take the first part of the echo (that which rises to the
+                    maximum) and add the decaying (FID like) part. This increases
+                    the SNR of the early points of the signal.
+    direct:         string
+                    Name of the direct dimension
+    exclude_rising: int
+                    In general it is assumed that the first few points of signal
+                    might be messed up due to dead time or other issues. This option
+                    allows us to add a rising edge to the echo and exclude the first
+                    few points of the signal. Note, to use this, add_rising must be
+                    True. 
+    slice_multiplier:   int
+                        The calculated frequency slice is calculated by taking the
+                        center frequency and extending out to values that are included
+                        in the peak times this multiplier. Therefore the larger this 
+                        value the larger the frequency slice.
+    peak_lower_thresh:  float
+                        fraction of the signal intensity used in calculating the 
+                        frequency slice. The smaller the value, the larger 
+                        wider the slice. This is especially useful for noisy data
+                        where this value may need to be decreases in order to include
```

Comment:
This is not a good suggestion, I think. If it's noisy, you would want to raise this and raise the slice multiplier, right?

Reply:
no lowering this value increases the wide_range (also in using this with the notebook examples I found this to be true. Increasing this or using the default for noisy data results in the assert statement at line 266. But will add the increase to the slice multiplier

Reply:
Yes, but that's problematic because you're increasing the range by locking on to noise

Reply:
the lower threshold is multiplied by the max of the signal (e.g. the peak) assuming a low SNR whether that's lower signal or larger noise simply pushes the wide range limits out. Additionally, if the signal is convolved and simply has low signal and a very broad linewidth, decreasing this value ensures that the lower limits of the signal is included. Further this isn't the final autoslice - this is JUST for deciding the wide range which is fed into the filter_range runction.

Reply:
Thinking out loud so that you can correct me if I'm wrong: according to your later comment isn't this rather locking onto multiple peaks? e.g. if the noise is on scale with the signal and it considers that a peak? So lowering this threshold does not guarantee that it will actually select the signal